finetune/eval/normalization.py

```diff
@@ -0,0 +1,56 @@
+from enum import IntEnum
+import math
+
+
+class NormalizationId(IntEnum):
+    """Enumeration of normalization methods."""
+
+    # No normalization is applied and the raw score is used.
+    NONE = 0
+
+    # Normalizes between [0, 1] using an inverse exponential function.
+    INVERSE_EXPONENTIAL = 1
+
+
+def normalize_score(
+    score: float,
+    normalization_id: NormalizationId,
+    norm_kwargs: dict = {},
+) -> float:
+    """Normalizes a score based on the provided normalization method.
+
+    Args:
+        score (float): The raw score to normalize.
+        normalization_id (NormalizationId): The identifier of the normalization method to use.
+        norm_kwargs (dict): Keyword arguments to pass to the normalization method.
+
+    Returns:
+        float: The normalized score.
+    """
+    match normalization_id:
+        case NormalizationId.NONE:
+            return _normalize_none(score)
+        case NormalizationId.INVERSE_EXPONENTIAL:
+            return _normalize_inverse_exponential(score, **norm_kwargs)
+        case _:
+            raise ValueError(f"Unhandled normalization method {normalization_id}.")
+
+
+def _normalize_inverse_exponential(score: float, ceiling: float) -> float:
+    """Normalizes between [0, 1] using an inverse exponential function.
```

Comment:
Running this in a notebook I get that passing in score = .999 and ceiling = 1 that the result is 0.417.

That feels too harsh?

For reference 0.9 is 0.357 and 0.8 is 0.289.

We could introduce a scaling factor multiply the ceiling by but that still kind of compresses the specific scores a little too much. Will look into some alternatives if you agree on the current scaling being too harsh.